File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/repository/MockGameRoleRepository.java
package com.yahyeet.boardbook.model.mock.repository;

import com.yahyeet.boardbook.model.entity.GameRole;
import com.yahyeet.boardbook.model.repository.IGameRoleRepository;

import java.util.List;
import java.util.concurrent.CompletableFuture;
import java.util.stream.Collectors;

/**
 * Mock implementation of a game role repository
 */
public class MockGameRoleRepository extends AbstractMockRepository<GameRole> implements IGameRoleRepository {
	@Override
	public CompletableFuture<List<GameRole>> findRolesByTeamId(String id) {
		return CompletableFuture.supplyAsync(() ->
			mockDatabase
				.stream()
				.filter(role -> role.getTeam().getId().equals(id))
				.collect(Collectors.toList())
		);
	}
}

File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/repository/MockGameRoleRepository.kt
package com.yahyeet.boardbook.model.mock.repository

import com.yahyeet.boardbook.model.entity.GameRole
import com.yahyeet.boardbook.model.repository.IGameRoleRepository
import java.util.concurrent.CompletableFuture
import java.util.stream.Collectors

class MockGameRoleRepository : AbstractMockRepository<GameRole>(), IGameRoleRepository {
    override fun findRolesByTeamId(id: String): CompletableFuture<List<GameRole>> {
        return CompletableFuture.supplyAsync { mockDatabase.stream().filter { role -> role.team!!.id == id }.collect<List<GameRole>, Any>(Collectors.toList()) }
    }
}
